sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.Metrics.cs

```diff
@@ -52,14 +52,22 @@ public MonitoringDataPoint GetDataPoint()
             Dictionary<string, AccumulatedValues> metricAccumulators = CreateMetricAccumulators(_collectionConfiguration);
             LiveMetricsBuffer liveMetricsBuffer = new();
             DocumentBuffer filledBuffer = _documentBuffer.FlipDocumentBuffers();
+            IEnumerable<DocumentStream> documentStreams = _collectionConfiguration.DocumentStreams;
             foreach (var item in filledBuffer.ReadAllAndClear())
             {
                 // TODO: item.DocumentStreamIds = new List<string> { "" }; - Will add the identifier for the specific filtering rules (if applicable). See also "matchingDocumentStreamIds" in AI SDK.
 
-                dataPoint.Documents.Add(item);
+                DocumentIngress? telemetryDocument = null;
+
+                CollectionConfigurationError[] groupErrors;
 
                 if (item is Request request)
                 {
+                    telemetryDocument = CreateTelemetryDocument(request,
+                        documentStreams,
+                        documentStream => documentStream.RequestQuotaTracker,
+                        documentStream => documentStream.CheckFilters(request, out groupErrors),
+                        CreateUpdatedRequest);
```

Comment:
Something is wrong here. We already have a TelemetryDocument `item`. Why are we creating a new document?

Reply:
It's used for creating the same item with new `documentStreamIds`. (The `DocumentIngress.DocumentStreamIds` is readonly so I need to create a new instance.)

This does the request in line 58 of this file.

Reply:
Had discussion with @xiang17, he will refactor to check the existing item and filter on it.

Reply:
Updated it.

Reply:
Adding it back in #43546.

At one point I thought the `documentStreamIds` are discarded by `Request`'s constructor body, but actually it's passed to the base class's constructor at the right most of the very long line (scroll right in the code box below):

https://github.com/Azure/azure-sdk-for-net/blob/7d04683d350c0908dbc4907444e87a97a207d640/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Generated/Models/Request.cs#L30-L37